mm2src/coins/utxo/utxo_builder/utxo_arc_builder.rs

```diff
@@ -266,10 +268,50 @@ pub(crate) async fn block_header_utxo_loop<T: UtxoCommonOps>(
         let storage = client.block_headers_storage();
         let from_block_height = match storage.get_last_block_height().await {
             Ok(Some(height)) => height,
-            Ok(None) => spv_conf.block_header.height,
-            Err(e) => {
-                error!("Error {e:?} on getting the height of the last stored {ticker} header in DB!",);
-                sync_status_loop_handle.notify_on_temp_error(e.to_string());
+            Ok(None) => {
+                let height = spv_conf.block_header.height;
+                match client
+                    .blockchain_block_header(height)
+                    .map_to_mm_fut(UtxoRpcError::from)
+                    .compat()
+                    .await
+                {
+                    Ok(header_bytes) => {
+                        let mut reader = Reader::new(&header_bytes);
+                        let Ok(header) = reader.read() else {
+                            let err = format!("Error deserializing starting block header from rpc - coin: {ticker} height: \
+                            {height}");
+                            error!("{err}");
+                            sync_status_loop_handle.notify_on_temp_error(err);
+                            continue;
+                        };
+
+                        if let Err(err) = spv_conf.validate_rpc_starting_block_header(height, &header) {
+                            sync_status_loop_handle.notify_on_permanent_error(err);
+                            break;
+                        };
+
+                        if let Err(err) = storage
+                            .add_block_headers_to_storage(HashMap::from([(height, header)]))
+                            .await
+                        {
+                            error!("{err:?}");
+                            sync_status_loop_handle.notify_on_temp_error(err);
+                            continue;
+                        };
+                    },
+                    Err(err) => {
+                        error!("Error {err:?} on getting {ticker} block header for height {height} from rpc!");
+                        sync_status_loop_handle.notify_on_temp_error(err);
+                        continue;
+                    },
+                };
```

Comment:
This is a bit much to be added inside a match statement. Can you please make this a function?